### PR TITLE
VTCCA-6212 Interface tests for the is-correct-agent page

### PR DESCRIPTION
**Pull request by joeyymoore**

### CODE REVIEW

**Comment by WillBlakeney** [comment]
rebuild

**Review by WillBlakeney — APPROVED** [review]
LGTM 👍

File: it/controllers/ListYearsFeatureSwitchISpec.scala [hunk 1]
```diff
@@ -157,4 +195,60 @@ class ListYearsFeatureSwitchISpec extends ISpecBase {
...
+      post("/auth/authorise")
+        .willReturn {
+          aResponse.withStatus(OK).withBody("{}")
+        }
+    }
+
+    val authData: OwnerAuthResult = scenario match {
+      case "OneAuthOneAgent" => testOwnerAuthResult1
+      case "NoAuths" => testOwnerAuthResultNoProperties
+      case "OneAuthNoAgent" => ownerAuthResultWithOneAuthorisation
```
Comment (WillBlakeney) [inline it/controllers/ListYearsFeatureSwitchISpec.scala:227]: `case "OneAuthNoAgent" => ownerAuthResultWithOneAuthorisation`


`ownerAuthResultWithOneAuthorisation` has an agent but is testing when there are no agents

```  val ownerAuthResultWithOneAuthorisation = OwnerAuthResult(
    start = 1,
    size = 15,
    filterTotal = 1,
    total = 1,
    authorisations = Seq(OwnerAuthorisation(
      authorisationId = 4222211L,
      status = "APPROVED",
      submissionId = "PLSubId",
      uarn = 999000111L,
      address = "123, SOME ADDRESS, SOME TOWN, AB1 CD2",
      localAuthorityRef = "BAREF",
      agents = Seq(OwnerAuthAgent(
        authorisedPartyId = 12L,
        organisationId = 1L,
        organisationName = "Some Org name",
        agentCode = 1001
      ))
    ))
  )```

File: it/base/TestData.scala [hunk 2]
```diff
@@ -215,4 +215,50 @@ trait TestData {
...
+          OwnerAuthAgent(
+            authorisedPartyId = 1L,
+            organisationId = 1L,
+            organisationName = "Test Agent",
+            agentCode = 1002
+          )
+        )
+      )
+    )
+  )
```
[the remark this answers is not in the document]
Comment (joeyymoore) [reply it/base/TestData.scala:262]: Sorry that was me being to trusting that the original unit tests made sense - will update to make it more obvious that the tests are just with 0/1/multiple auths

**Pull request merged**